src/core/producer/stage.h

```diff
@@ -58,8 +58,10 @@ class stage final
 
     explicit stage(int channel_index, spl::shared_ptr<caspar::diagnostics::graph> graph);
 
-    std::map<int, layer_frame>
-    operator()(const video_format_desc& format_desc, int nb_samples, std::vector<int>& fetch_background);
+    std::map<int, layer_frame> operator()(const video_format_desc& format_desc,
```

Comment:
the return type of this should be changed to use draw_frame instead of layer_frame.
It was changed to layer_frame for routes to use, so with that inside the stage this complexity can be removed

Reply:
Agreed, done. layer_frame is now only used for the route callback.